### PR TITLE
GithubService, GithubSearch and GithubRepo classes

### PR DESCRIPTION
**Pull request by dborski**
## Spec/Features ##
* Create GithubService class and spec
  - This service makes our API request to the Github API and grabs all repos for the user and returns it in hash form
* Create GithubSearch class and spec
  - This class takes our repos hash and creates GitubRepo POROs
* Create GithubRepo class and spec
  - This class defines our GithubRepo objects and attributes

### CODE REVIEW

File: spec/poros/github_repo_spec.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  expect(repo).to be_a(GithubRepo)
+  end
+
+  it 'attributes' do
+  repo = GithubRepo.new({name: "brownfield-of-dreams", html_url: "https://github.com/dborski/brownfield-of-dreams"})
+
+  expect(repo.name).to eq("brownfield-of-dreams")
+  expect(repo.url).to eq("https://github.com/dborski/brownfield-of-dreams")
+  end
+end
```
Comment (dborski) [inline spec/poros/github_repo_spec.rb:16]: Will need to make test more robust in future

File: spec/services/github_search_spec.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    search = GithubSearch.new
+    token = ENV['GITHUB_API_KEY']
+
+
+    repos = search.repos(token)
+
+    expect(repos.first).to be_a(GithubRepo)
+    expect(repos.first.name).to eq("brownfield-of-dreams")
+  end
+end
```
Comment (dborski) [inline spec/services/github_search_spec.rb:14]: Need to use Webmock here in the future so we're not making an actual API call

File: spec/services/github_service_spec.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  it 'sends repos for a single user' do
+    token = ENV['GITHUB_API_KEY']
+    github = GithubService.new
+    repos = github.get_repos(token)
+
+
+    expect(repos.first[:name]).to eq("brownfield-of-dreams")
+    expect(repos.count).to eq(30)
+  end
+end
```
Comment (dborski) [inline spec/services/github_service_spec.rb:13]: Need to use Webmock here in the future so we're not making an actual API call

File: app/models/github_search.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+class GithubSearch
+
```
Comment (Shock253) [inline app/models/github_search.rb:2]: We might have to rename/change this to a facade later

File: app/services/github_service.rb [hunk 5]
```diff
@@ -0,0 +1,23 @@
+class GithubService
+
+  def get_repos(github_token)
+    params = {sort: "created", direction: "desc"}
+
+    get_json('user/repos', params, github_token)
+  end 
+
```
Comment (Shock253) [inline app/services/github_service.rb:8]: I'll just be adding new methods here

File: spec/poros/github_repo_spec.rb (hunk 1 above)
Comment (Shock253) [reply spec/poros/github_repo_spec.rb:16]: And add some api mocking using vcr/webmock

**Review by Shock253 — APPROVED** [review]

**Pull request merged**